### PR TITLE
Update StatsBase compat

### PR DESCRIPTION
**Pull request by akshaysridhar**
Updates compat restriction on `StatsBase`.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -55,7 +55,6 @@ LinearAlgebra = "1"
 NCDatasets = "0.11, 0.12, 0.13, 0.14"
 SciMLBase = "1, 2"
 StaticArrays = "1"
-StatsBase = "0.34"
```
Comment (nefrathenrici) [inline Project.toml:58]: I think the compat should be `StatsBase = "0.33, 0.34"`. Atmos has the same [compat](https://github.com/CliMA/ClimaAtmos.jl/blob/main/Project.toml#L80C1-L80C25)

Comment (akshaysridhar) [reply Project.toml:58]: Fixed.

**Comment by nefrathenrici** [comment]
Should we make a patch release with the updated compat?

**Comment by juliasloan25** [comment]
> Should we make a patch release with the updated compat?

Sure, we can! I was hoping to make patch releases of ClimaLand and ClimaAtmos today, so if we can get this in soon we can make a release with it

**Review by juliasloan25 — APPROVED** [review]

**Pull request merged**